apollo-runtime/src/main/java/com/apollographql/apollo/ApolloClient.java

```diff
@@ -157,6 +153,26 @@ Response cachedHttpResponse(String cacheKey) throws IOException {
     }
   }
 
+  private <D extends Operation.Data, T, V extends Operation.Variables> RealApolloCall<T> newCall(
+      @Nonnull Operation<D, T, V> operation) {
+    ResponseFieldMapper responseFieldMapper = responseFieldMapper(operation);
+    return new RealApolloCall<T>(operation, serverUrl, httpCallFactory, httpCache, defaultHttpCachePolicy, moshi,
```

Comment:
this constructor is wild! is there a way we can encapsulate somehow.  Of the top of my head I see 2 options: builder or organization of params into a few objects.